collections/slice.go

```diff
@@ -1,9 +1,33 @@
 package collections
 
-func TransformSlice[TInput any, TOutput any](slice []TInput, transformFn func(TInput) TOutput) []TOutput {
+func All[T any](slice []T, predicate func(T) bool) bool {
+	negatedPredicate := func(val T) bool {
+		return !predicate(val)
+	}
+	return !Any(slice, negatedPredicate)
+}
+
+func Any[T any](slice []T, predicate func(T) bool) bool {
+	for _, val := range slice {
+		if predicate(val) {
+			return true
+		}
+	}
+	return false
+}
+
+func Transform[TInput any, TOutput any](slice []TInput, transformFn func(TInput) TOutput) []TOutput {
 	var result []TOutput
 	for _, val := range slice {
 		result = append(result, transformFn(val))
 	}
 	return result
 }
+
+func ToMap[TKey comparable, TValue any](slice []TValue, keyFn func(TValue) TKey) map[TKey]TValue {
+	result := make(map[TKey]TValue)
+	for _, val := range slice {
```

Comment:
Do you need to copy the val to avoid the slice iterator bug? I think you're OK because you're not taking a pointer to the value, but wasn't 100% sure it was safe with generics.